File: ts/src/80.ts
//1752. Check if Array Is Sorted and Rotated

function check(nums: number[]): boolean {
  let countBreaks = 0;
  const n = nums.length;

  for (let i = 0; i < n; i++) {
    if (nums[i] > nums[(i + 1) % n]) {
      countBreaks++;
    }
  }

  return countBreaks <= 1;
}

//sol:
//https://leetcode.com/problems/check-if-array-is-sorted-and-rotated/solutions/6364065/1752-check-if-array-is-sorted-and-rotate-dn3t/

File: rust/src/80.rs
//80. Remove Duplicates from Sorted Array II
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        if nums.len() <= 2 {
            return nums.len().try_into().unwrap();
        }

        let mut idx = 2;

        for i in 2..nums.len() {
            if nums[i] != nums[idx - 2] {
                nums[idx] = nums[i];
                idx += 1;
            }
        }

        idx.try_into().unwrap()
    }
}
//sub: https://leetcode.com/problems/remove-duplicates-from-sorted-array-ii/submissions/1606135466/?envType=study-plan-v2&envId=top-interview-150
//sol: https://leetcode.com/problems/remove-duplicates-from-sorted-array-ii/solutions/6648352/80-remove-duplicates-from-sorted-array-i-dl4x/
